mmf/common/test_reporter.py

```diff
@@ -95,6 +98,10 @@ def flush_report(self):
         logger.info(f"Wrote predictions for {name} to {os.path.abspath(filepath)}")
         self.report = []
 
+    def postprocess_dataset_report(self):
+        if hasattr(self.current_dataset, "on_prediction_end"):
+            self.report = self.current_dataset.on_prediction_end(self.report)
```

Comment:
@vedanuj What do you think is the right way to do this? Should this be handled via callbacks?

Reply:
The right way to do this will be to add a new Reporting callback, with hooks `on_validation_end` or `on_prediction_end` as required. However this might be fine for now given we are moving to lightning soon. 

cc @ytsheng Does Lightning have any callbacks for report generation or report gathering after validation/test?